go/libraries/doltcore/sqle/tables.go

```diff
@@ -1201,6 +1207,70 @@ func (t *AlterableDoltTable) CreateForeignKey(
 	return t.updateFromRoot(ctx, newRoot)
 }
 
+// addEmptyForeignKeyToCollection creates a foreign key with delayed resolution. A FK with delayed resolution
+// does not have a referencedTableIndex nor a ReferencesTable Column.
+func (t *AlterableDoltTable) addEmptyForeignKeyToCollection(ctx *sql.Context, root *doltdb.RootValue, fkName, tableIndexName string, colTags []uint64, refTblName string, refTblColNames []string, onUpdate, onDelete sql.ForeignKeyReferenceOption) error {
+	onUpdateRefOp, err := parseFkReferenceOption(onUpdate)
+	if err != nil {
+		return err
+	}
+	onDeleteRefOp, err := parseFkReferenceOption(onDelete)
+	if err != nil {
+		return err
+	}
+
+	foreignKey := doltdb.ForeignKey{
+		Name:                       fkName,
+		TableName:                  t.name,
+		TableIndex:                 tableIndexName,
+		TableColumns:               colTags,
+		ReferencedTableName:        refTblName,
+		ReferencedTableIndex:       "",
+		ReferencedTableColumns:     nil,
+		ReferencedTableColumnNames: refTblColNames,
+		OnUpdate:                   onUpdateRefOp,
+		OnDelete:                   onDeleteRefOp,
+	}
+
+	foreignKeyCollection, err := root.GetForeignKeyCollection(ctx)
```

Comment:
With my earlier comment about first writing all foreign keys as "empty", you should remove writing to the root here and do it in the calling function.